### ISSUE TITLE
Invalid memory accesses in calc_coupling_elem_spin0and2_pure?

### ISSUE THREAD

**Issue by mreineck** [simonsobs/pspy#80]
While running benchmarks on the code I noticed a block in hte Fortran sources that looks dangerous to me: https://github.com/simonsobs/pspy/blob/1d5d8734f489c6a15fe6ee53d8d61b25478fd9b4/pspy/mcm_fortran/mcm_fortran.f90#L77C1-L86C1
```Fortran
        if (i2 < 0) then
            fac_b = 0d0
        end if

        if (i3 < 0) then
            fac_c = 0d0
        end if

        combin = thrcofa(i1) + fac_b * thrcofb(i2) + fac_c * thrcofc(i3)
```
Here, `i2` and `i3` can obviously take values below 1, but they are used to access the arrays `thrcofb` and `thrcofc` whose first index is 1. This is undefined behaviour and can cause segmentation faults.
Even if this currently doesn't cause any issues, I'd still suggest to avoid these accesses, since they will make debugging of potential other problems much harder.

**Comment by xgarrido** [comment]
I think a correct way to handle these cases is the following 
```fortran
        if (i2 < 1) then
            combin2 = 0d0
         else
            combin2 = fac_b * thrcofb(i2)
        end if

        if (i3 < 1) then
            combin3 = 0d0
         else
            combin3 = fac_c * thrcofc(i3)
        end if

        combin = thrcofa(i1) + combin2 + combin3
```
where `combin2` and `combin3` are `real(8)`. I've added a test for pure BB mode with comparisons with NaMaster (see PR #82). The above code (not yet implemented in PR #82) is still "physically" correct but prevent access to undefined array values. Does it look fine for you ?

**Comment by mreineck** [comment]
Looks great to me, thanks!
